src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,179 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
+    {
+        public TextLoaderRange()
+        {
+        }
+
+        /// <summary>
+        /// Convenience constructor for the scalar case, when a given column 
+        /// in the schema spans only a single column in the dataset.
+        /// <see cref="Min"/> and <see cref="Max"/> are set to the single value <paramref name="ordinal"/>.
+        /// </summary>
+        /// <param name="ordinal">Column index in the dataset.</param>
+        public TextLoaderRange(int ordinal)
+        {
+
+            Contracts.CheckParam(ordinal >= 0, nameof(ordinal), "Cannot be a negative number");
+
+            Min = ordinal;
+            Max = ordinal;
+        }
+
+        /// <summary>
+        /// Convenience constructor for the vector case, when a given column 
+        /// in the schema spans contiguous columns in the dataset.
+        /// </summary>
+        /// <param name="min">Starting column index in the dataset.</param>
+        /// <param name="max">Ending column index in the dataset.</param>
+        public TextLoaderRange(int min, int max)
+        {
+
+            Contracts.CheckParam(min >= 0, nameof(min), "Cannot be negative number.");
+            Contracts.CheckParam(max >= min, nameof(max), $"Cannot be less than {nameof(min)}.");
```

Comment:
>$"Cannot be less than {nameof(min)}." [](start = 58, length = 37)

Note that this `$` interpolation means this isn't a constant. Note however, that `"Cannot be less than " + nameof(min) + "."` would be a constant, and so we could use it in a `Check` error message.